src/metrics-gatherer.ts

```diff
@@ -40,90 +41,138 @@ export class MetricsGatherer {
 	}
 
 	private initState() {
-		this.metrics = {
-			gauge: {},
-			counter: {},
-			summary: {},
-			histogram: {},
-		};
-		this.customParams = {};
-		this.descriptions = {};
-		this.existMap = {};
-		this.kinds = {};
+		try {
+			this.metrics = {
+				gauge: {},
+				counter: {},
+				summary: {},
+				histogram: {},
+			};
+			this.customParams = {};
+			this.descriptions = {};
+			this.existMap = {};
+			this.kinds = {};
+			this.internalErrorCount = 0;
+		} catch (e) {
+			this.err(e);
+		}
 	}
 
 	// fetch the description for a metric
 	public describe(name: string, text: string, custom: CustomParams = {}) {
-		if (this.descriptions[name]) {
-			throw new MetricsGathererError(
-				`tried to describe metric "${name}" twice`,
-			);
+		try {
+			if (this.descriptions[name]) {
+				this.err(
+					new MetricsGathererError(`tried to describe metric "${name}" twice`),
```

Comment:
Might be worth just throwing this and handling it with everything else in the `catch` below